### PR TITLE
feat: ERC721 tests & fixes

### LINKED ISSUE #744: [ERC721] Write test suite
Write a comprehensive test suite for the erc721 implementation here: https://github.com/dojoengine/dojo/tree/main/crates/dojo-erc/src/erc721

Inspiration can be taken from the [Open Zeppelin teams](https://github.com/OpenZeppelin/cairo-contracts/tree/cairo-2) test suites

### PR DESCRIPTION
**Pull request by notV4l**
- partially closes #744 

add ERC721 tests
fix ERC721 issues

still need more tests

### CODE REVIEW

File: crates/dojo-erc/src/erc721/erc721.cairo [hunk 1]
```diff
@@ -118,29 +143,75 @@ mod ERC721 {
...
+        self.emit(ApprovalForAll { owner: caller, operator, approved });
+    }
+
+    #[external(v0)]
+    fn transfer(ref self: ContractState, to: ContractAddress, token_id: u256) {
+        transfer_from(ref self, get_caller_address(), to, token_id);
+    }
+
+    #[external(v0)]
+    fn safe_transfer_from(
```
Comment (tarrencev) [inline crates/dojo-erc/src/erc721/erc721.cairo:164]: safe transfer isn't needed with account abstraction. let's remove this method. see: https://github.com/starknet-io/SNIPs/discussions/36#discussioncomment-6640087

File: crates/dojo-erc/src/erc721/erc721.cairo [hunk 2]
```diff
@@ -15,9 +15,10 @@ mod ERC721 {
     #[storage]
     struct Storage {
         world: IWorldDispatcher,
-        token_name: felt252,
-        token_symbol: felt252,
-        token_uri: felt252,
+        _owner: ContractAddress,
```
[the remark this answers is not in the document]
Comment (notV4l) [reply crates/dojo-erc/src/erc721/erc721.cairo]: i only see cairo code XD

File: crates/dojo-erc/src/erc721/erc721.cairo [hunk 3]
```diff
@@ -51,48 +52,68 @@ mod ERC721 {
...
     fn symbol(self: @ContractState) -> felt252 {
-        self.token_symbol.read()
+        self._symbol.read()
     }
 
     #[external(v0)]
-    fn uri(self: @ContractState) -> felt252 {
-        self.token_uri.read()
+    fn token_uri(self: @ContractState, token_id: u256) -> felt252 {
+        assert(exists(self, token_id), 'ERC721: invalid token_id');
```
[the remark this answers is not in the document]
Comment (notV4l) [reply crates/dojo-erc/src/erc721/erc721.cairo]: removed to keep it simple

File: crates/dojo-erc/src/erc721/erc721.cairo (hunk 1 above)
Comment (notV4l) [reply crates/dojo-erc/src/erc721/erc721.cairo]: what happen if you want to send an nft to a contract & run some code on transfer ?

[IERC721Receiver.onERC721Received](https://docs.openzeppelin.com/contracts/4.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-)

**Pull request merged**